lib/wor/paginate/utils/preserve_records_helper.rb

```diff
@@ -0,0 +1,64 @@
+require_relative 'uri_helper'
+
+module Wor
+  module Paginate
+    module Utils
+      class PreserveRecordsHelper
+        def initialize(content, url, options)
+          @content = content
+          @url = url
+          @options = options
+        end
+
+        def call
+          [content.where("#{field} <= :last_value", last_value: last_value),
+           UriHelper.replace_query_params(url, query_param_name => last_value)]
+        end
+
+        private
+
+        attr_reader :content, :url, :options
+
+        def by
+          @by ||= begin
+            by = options[:by]&.to_sym || :timestamp
+            raise ArgumentError, "'by' option shuld be 'id' or 'timestamp'" unless
+              %i[timestamp id].include? by
+
+            by
+          end
+        end
+
+        def field
+          @field ||= options[:field] || case by
```

Comment:
Not a correction because it's not wrong, but I'm not a huge fan of `case`s in the right side. Maybe delegate?